src/harness/dynamicUnittestsImpl.ts

```diff
@@ -0,0 +1,73 @@
+
+namespace DynamicUnittests {
+    export const enum TestType {
+        Tsbuild = "tsbuild",
+        TsbuildWatch = "tsbuildWatch",
+        Tsc = "tsc",
+        TscWatch = "tscWatch",
+        Tsserver = "tsserver",
+    }
+    export function runTests(basePath: string, fileName: string, dynamicTestName: string) {
+        const content = Harness.IO.readFile(fileName)!;
+
+        // Give file paths an absolute path for the virtual file system
+        const absoluteBasePath = ts.combinePaths(Harness.virtualFileSystemRoot, basePath);
+        const absoluteFileName = ts.combinePaths(Harness.virtualFileSystemRoot, fileName);
+
+        // Parse out the files and their metadata
+        // const testData = parseTestData(absoluteBasePath, content, absoluteFileName);
+        // const state = new TestState(absoluteFileName, absoluteBasePath, testType, testData);
+        const actualFileName = Harness.IO.resolvePath(fileName) || absoluteFileName;
+        const output = ts.transpileModule(content, { reportDiagnostics: true, fileName: actualFileName, compilerOptions: { target: ts.ScriptTarget.ES2015, inlineSourceMap: true, inlineSources: true } });

```

Comment:
I'm not sure that this is going to be desirable; after modules, the tests compile fully in like 200ms and start running, but this would mean that we completely defer compiling modules that actually are tests that would normally run in the test runner, sort of defeating the speedup.

I understand doing this for tests like the compiler baselines, where the thing being tested is the compiler itself, or for fourslash where the syntax is entirely different, but I don't think that we should defer these tests like this and then run them as though they were being executed in `run.js`.

Reply:
There will be a `run.js`, but `tsc` won't actually have been run to produce it, so the d.ts files aren't guaranteed to exist.

I don't mind this PR in general if it were to run kind of like a fourslash test with a constrained interface that's declared in a `d.ts` file, I just don't want to depend on having built the project for type checking to work in these files (because they may not exist; that technically would happen already if you opened the editor in this PR and had not yet built the repo).

Reply:
Created file with references to ts files so it works without having to generate d.ts file for `run`